internal/venice-client-common/src/main/java/com/linkedin/venice/compute/ComputeRequestWrapper.java

```diff
@@ -153,4 +156,20 @@ public void setOperations(List<ComputeOperation> operations) {
         throw new VeniceException("Compute request version " + version + " is not support yet.");
     }
   }
+
+  public void initializeOperationResultFields(Schema resultSchema) {
+    List<ComputeOperation> operations = getOperations();
+    this.operationResultFields = new ArrayList<>(operations.size());
+    ComputeOperation computeOperation;
+    ReadComputeOperator operator;
+    for (int i = 0; i < operations.size(); i++) {
```

Comment:
Shall we use an iterator here in case the `operations` is using a LinkedList-like structure?

Reply:
Hmm... should we worry about that? My understanding is that LL almost always underperforms ArrayList...

Reply:
Correct, the iterator is mainly for defensive purpose in case someone changes the implementation to use a LinkedList.